packages/flutter/lib/src/material/slider.dart

```diff
@@ -1650,7 +1650,22 @@ class _RenderSlider extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
       sliderTheme: _sliderTheme,
       isDiscrete: isDiscrete,
     );
-    final Offset thumbCenter = Offset(trackRect.left + visualPosition * trackRect.width, trackRect.center.dy);
+    final double padding = isDiscrete || _sliderTheme.trackShape is RoundedRectSliderTrackShape ? trackRect.height : 0.0;
```

Comment:
Do we use `is` a lot in the slider code like this?

Reply:
Maybe instead of `is`, `RoundedRectSliderTrackShape` and others can contains a getter for this?

Reply:
Excellent idea! Added `isRounded` getter. Please let me know what do you think.